wasm/wat/parser.go

```diff
@@ -0,0 +1,230 @@
+package wat
+
+import (
+	"fmt"
+)
+
+type ModuleParser struct {
+	source                     []byte
+	m                          *textModule
+	parenDepth, skipUntilDepth int
+	// currentStringCount allows us to unquote the Import.module and Import.name fields, differentiating empty from
+	// never set, without making Import.module and Import.name pointers.
+	currentStringCount int
+	pf                 fieldHandler // TODO: crappy name
```

Comment:
I will address this and the below TODO prior to marking for review (other TODO will remain to keep line count down)